src/core.rs

```diff
@@ -427,3 +427,20 @@ pub enum NormType {
     /// Normalized using minmax distance
     MinMax = 32,
 }
+
+/// Threshold type. Please refer to [OpenCV's
+/// documentation] (https://docs.opencv.org/3.4.0/d7/d1b/group__imgproc__misc.html#gaa9e58d2860d4afa658ef70a9b1115576)
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum ThreshType {
```

Comment:
Some suggestions:

1. s/ThreshType/ThresholdType
2. Use OpenCV trunk URL?
https://docs.opencv.org/trunk/d7/d1b/group__imgproc__misc.html#gaa9e58d2860d4afa658ef70a9b1115576
3. Each enum item can omit the prefix "THRESH". In this way, users would use `ThresholdType::Binary`, more clear than `ThreshType::ThreshBinary`.
4. Can you include all types? (and I think THRESH_BINARY is 0, not 1).

Reply:
Hey, thanks for the fast reply. Yea sorry, my bad regarding Threshhold types. Will change asap, aslo looking at the trunk URL.. should I add "Mask, Otsu and Triangle' flags to the enum as well. Cannot find what int's correspond to them. Probably just 5, 6, 7?

Reply:
I think they are 7, 8, 16: https://github.com/opencv/opencv/blob/6e13ffe162b1e6ec79bcbea0dd609f33aef23051/modules/imgproc/include/opencv2/imgproc.hpp#L332

```
cv::ThresholdTypes { 
  cv::THRESH_BINARY = 0, 
  cv::THRESH_BINARY_INV = 1, 
  cv::THRESH_TRUNC = 2, 
  cv::THRESH_TOZERO = 3, 
  cv::THRESH_TOZERO_INV = 4, 
  cv::THRESH_MASK = 7, 
  cv::THRESH_OTSU = 8, 
  cv::THRESH_TRIANGLE = 16 
}
```